### PR TITLE
Enable access to the full genbank record

### PR DESCRIPTION
**Pull request by emckee2006**

### CODE REVIEW

File: biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java [hunk 1]
```diff
@@ -440,4 +442,22 @@ public void parseFeatures(AbstractSequence<C> sequence) {
 	public CompoundSet<?> getCompoundType() {
 		return compoundType;
 	}
+
+
+
+	/**
+	 * @return the srcRecord
+	 */
+	public String getSrcRecord() {
```
Comment (josemduarte) [inline biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java:451]: It'd help to have javadocs here. What exactly is the source record? The whole genbank file? a part of it? what format?

File: biojava-core/src/main/java/org/biojava/nbio/core/sequence/template/AbstractSequence.java [hunk 2]
```diff
@@ -732,5 +734,18 @@ public int countCompounds(C... compounds) {
 		return SequenceMixin.inverse(this);
 	}
 
+	/**
+	 * @return the srcRecord
+	 */
+	public String getSrcRecord() {
```
Comment (josemduarte) [inline biojava-core/src/main/java/org/biojava/nbio/core/sequence/template/AbstractSequence.java:740]: Same comment about javadocs

**Comment by sbliven** [comment]
What's the memory footprint of this change? Does it double the storage size?

File: biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java (hunk 1 above)
Comment (sbliven) [reply biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankSequenceParser.java:452]: Thou hast a typo :)

**Comment by sbliven** [comment]
What's the use case for this? It seems very strange to me for a parser to store the unparsed input.

**Comment by emckee2006** [comment]
I must be missing the typo...

**Comment by emckee2006** [comment]
We are stepping through the refseq/genbank release files one record at a
time. We do some checksumming and such in an attempt to determine if the
record has been modified since we last loaded it to our database...

**Comment by heuermh** [comment]
> What's the use case for this? It seems very strange to me for a parser to store the unparsed input.

I would have to agree here.

Is it possible for this use case to break the parser up into a record splitter (which one could use to checksum the entire record text) and a record tokenizer (which then parses the record into the java objects)?

**Comment by emckee2006** [comment]
Or add in a flag to turn off the full record storing?

**Comment by sbliven** [comment]
I'd suggest leaving this out of biojava. I think you can implement a splitter class as @heuermh suggested, and then use biojava to parse the split records.

**Comment by emckee2006** [comment]
I'm planning on something like that... it does sound like a better
approach.

**Comment by josemduarte** [comment]
Thanks @emckee2006 and thanks all for reviewing. I'll close then

**Pull request closed without merging**